mobile/src/main/java/ch/epfl/sdp/mobile/ui/authentication/AuthenticationScreen.kt

```diff
@@ -72,7 +72,7 @@ fun AuthenticationScreen(
         ) {
           TextField(
               value = state.name,
-              onValueChange = { state.name = it },
+              onValueChange = { state.name = it.trim() },
```

Comment:
With this, we cannot have a username with a space inside (like "Chau Ying")

Reply:
Not really, trim only removes leading and trailing white spaces, it's actually equivalent to `name.trimStart().trimEnd()`. So `"Chau Ying".trim()` will result in `"Chau Ying"` as expected and no words will be omitted.

Reply:
But if i'm not mistaked, since it is `onValueChange`, you will never be able to write the ` space` character, except if you write the name without spaces and then click where you wanted to insert it initially.

Reply:
@matt989253 and @KurohanaJuri are right - applying the trim in `onValueChange` restricts what users can type; trimming the names should be applied only when submitting the values and thus creating the account, and could (for instance) be performed in the facade directly.

Reply:
I would prefer to trim the inputs when making the calls to the concerned facade methods in the stateful authentication screen.